test/models/department_thesis_test.rb

```diff
@@ -0,0 +1,42 @@
+# == Schema Information
+#
+# Table name: department_theses
+#
+#  thesis_id     :integer
+#  department_id :integer
+#  id            :integer          not null, primary key
+#  primary       :boolean          default(FALSE), not null
+#
+
+require 'test_helper'
+
+class DepartmentThesisTest < ActiveSupport::TestCase
+  test 'valid link with and without primary' do
+    link = department_theses(:primary)
+    assert_equal link.primary, true
+    assert(link.valid?)
+    link = department_theses(:other)
+    assert_equal link.primary, false
+    assert(link.valid?)
+  end
+
+  test 'only one primary department per thesis' do
```

Comment:
Looking at this test doesn't make me confident without looking at the underlying data that the test does what it says. Would you consider making it clearer in the test what is going on?

Reply:
Sure - looking at this again with fresh coffee, I can see that this is a bit sparse. I've got a longer test now that does more to explicitly test the underlying conditions prior to these actions.